packages/smart-wallet/src/typeGuards.js

```diff
@@ -0,0 +1,66 @@
+// @ts-check
+
+import { M } from '@agoric/store';
+import {
+  InstanceHandleShape,
+  ProposalShape,
+} from '@agoric/zoe/src/typeGuards.js';
+
+export const shape = {
+  // smartWallet
+  StringCapData: {
+    body: M.string(),
+    slots: M.arrayOf(M.string()),
+  },
+
+  // invitations
+  ContractInvitationSpec: M.split(
+    {
+      source: M.string(), // 'contract
+      instance: InstanceHandleShape,
+      publicInvitationMaker: M.string(),
+    },
+    M.partial({
+      invitationArgs: M.array(),
+    }),
+  ),
+  ContinuingInvitationSpec: M.split(
+    {
+      source: M.string(), // 'continuing'
```

Comment:
@erights I didn't see a way in the patterns lib to express a literal. I saw some "limits" feature on some matchers like string but it's all numerical.

WDYT of `M.literal(val)` ?